### PR TITLE
fix(bookstore): incorrect img source for booksCompany

### PR DESCRIPTION
**Pull request by YuanLiou**
#### Update:
  0103: Use regex to get the proper image from data-src.

#### Change:
  - Update the quering field of the book image from BookCompany.
  - Due to fields change, we can't get the correct image.
  - We'll get the url of this **i** value: getImage?i=https://www.books.com.tw/img/E05/001/27/E050012722.jpg

source sample:
```html
<div class="box_1">
	<i class="sign_ebook"></i>
	<a target="_blank"
		href="//search.books.com.tw/redirect/move/key/%E9%AB%98%E6%95%88%E4%BA%BA%E7%94%9F%E7%9A%84%E6%B8%85%E5%96%AE%E6%95%B4%E7%90%86%E8%A1%93/area/mid/item/E050012722/page/1/idx/5/cat/E05/pdf/1/spell/1"
		title="Evernote 100個做筆記的好方法：數位化重整你的工作與人生(全新增訂版) (電子書)"><img class="b-lazy"
			src="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
			data-src="https://im1.book.com.tw/image/getImage?i=https://www.books.com.tw/img/E05/001/27/E050012722.jpg&amp;w=187&amp;h=187&amp;v=59a54b4c"
			data-srcset="https://im1.book.com.tw/image/getImage?i=https://www.books.com.tw/img/E05/001/27/E050012722.jpg&amp;w=374&amp;h=374&amp;v=59a54b4c 2x"
			alt="Evernote 100個做筆記的好方法：數位化重整你的工作與人生(全新增訂版) (電子書)"></a>
```
     - Change the target fields to: .box_1 class -> a -> img -> data-src field.

#### Preview:

| before | after |
| ---- | ---- |
|||

### CODE REVIEW

**Comment by YuanLiou** [comment]
用正則去取 getImage？i= 後面的網址，例如：https://www.books.com.tw/img/E05/001/27/E050012722.jpg

File: src/stores/booksCompany.ts [hunk 1]
```diff
@@ -141,3 +147,10 @@ function _getBooks($: CheerioAPI) {
 
   return books;
 }
+
+function _getBooksCompanyThumbnail(url: string) {
+  const thumbnailRegexPattern = /.+\/getImage\?i=(https:\/\/.+\.jpg).+/;
```
Comment (YuanLiou) [inline src/stores/booksCompany.ts:152]: Test from [here, regex101.com](https://regex101.com/r/3KHwFy/1)

Comment (YuerLee) [reply src/stores/booksCompany.ts:152]: Nice.

File: src/stores/booksCompany.ts [hunk 2]
```diff
@@ -120,9 +120,15 @@ function _getBooks($: CheerioAPI) {
         .replace(/NT\$|,/g, ''),
     );
 
+    const rawThumbnailUrl = $('.box_1', elem)
+      .children('a')
+      .children('img')
+      .prop('data-src') as string;
```
Comment (YuerLee) [inline src/stores/booksCompany.ts]: 預先宣告型態 `const rawThumbnailUrl: string = xxx` 可能比強制指定型態更好。

But，有時其值就是得強制轉型。

Comment (YuanLiou) [reply src/stores/booksCompany.ts]: 我來更新下，謝謝提醒。

**Review by YuerLee — APPROVED** [review]
帥

**Pull request merged**